### DISCUSSION TITLE
Getting more information from button_handler

### DISCUSSION THREAD

**Discussion by ghost** [beeware/toga#1379] [Q&A] [answered]
I have a grid of buttons constructed using boxes (I think this is the right way to go about this in Toga, but not entirely sure). Is there a way for me to determine using the on_press which button was pressed?  They are displayed using a for loop class. Think something like minesweeper.

Thanks!

**Comment by ghost** [comment, marked as answer]
I figured out one way was to keep the id along with the row and column in a list of dictionaries and look it up from widget.id in the callback method, if anyone else is stupid enough not to know a solution. ;)

**Reply by freakboy3742** [reply]
Glad you worked it out; FWIW, you can use the widget itself (rather than the widget ID) as lookup key as well.

**Reply by ghost** [reply]
Thanks, I'm not very clever when it comes to such things. I usually end up doing things the hard way.
